libnetwork/internal/setmatrix/setmatrix.go

```diff
@@ -7,36 +7,30 @@ import (
 )
 
 // SetMatrix is a map of Sets.
-type SetMatrix struct {
+// The zero value is an empty set matrix ready to use.
+type SetMatrix[T comparable] struct {
 	matrix map[string]mapset.Set
 
 	mu sync.Mutex
 }
 
-// NewSetMatrix creates a new set matrix object.
-func NewSetMatrix() *SetMatrix {
-	s := &SetMatrix{}
-	s.init()
-	return s
-}
-
-func (s *SetMatrix) init() {
-	s.matrix = make(map[string]mapset.Set)
-}
-
 // Get returns the members of the set for a specific key as a slice.
-func (s *SetMatrix) Get(key string) ([]interface{}, bool) {
+func (s *SetMatrix[T]) Get(key string) ([]T, bool) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	set, ok := s.matrix[key]
 	if !ok {
 		return nil, ok
 	}
-	return set.ToSlice(), ok
+	slice := make([]T, 0, set.Cardinality())
```

Comment:
Because an `[]interface{}` value is not assignable to a variable of type `[]T`

Reply:
Ah, I see: we're still using an old version of `golang-set`. It supports generics since its v2.0 and `ToSlice()` now returns `[]T`. Maybe we should update it?

Reply:
Good catch! I completely overlooked that there was a v2 module